test/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+add_subdirectory(includes)
+add_subdirectory(functional/fixtures)
+
+if(POLICY CMP0012)
+  # Handle CI=true, without dev warnings.
+  cmake_policy(SET CMP0012 NEW)
+endif()
+
+find_program(NVIM_PRG nvim)
+if(NOT NVIM_PRG)
+  set(NVIM_PRG $<TARGET_FILE:nvim>)
+endif()
+
+set(test_env)
+list(
+  APPEND test_env
+  "LC_ALL=en_US.UTF-8"
+)
+
+add_test(
+  NAME lintcommit
+  COMMAND ${NVIM_PRG} -u NONE --headless -es -c [==[lua require('scripts.lintcommit').main({trace=false})]==]
+  WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+)
+
+get_target_property(NVIM_SOURCES nvim_sources INTERFACE_SOURCES)
+get_target_property(NVIM_HEADERS nvim_headers INTERFACE_SOURCES)
```

Comment:
OOF, this one hurts bad. These variables were relatively self-contained before, now we're defining these in one subdirectory and using it in another, which kinda breaks the purpose of subdirectories IMO. We might need to rethink this.

Do we really need to move these to `test/`? I mean it makes sense that tests are located in `test/`, but maybe we should just replace the lint targets with tests on their current location?

Reply:
we can create a function that will add a bunch of tests for a list of source files, what do you think?

Reply:
Could work? I dunno, feels easier to just use `add_test` IMO.

Reply:
Eh, sure, a function is fine if that's easier.